pkg/products/marin3r/rateLimitService.go

```diff
@@ -392,3 +420,117 @@ func GetSecondsInUnit(seconds uint64) (string, error) {
 		return "", fmt.Errorf("unexpected seconds value: %v, while getting seconds in Rate Limit Unit", seconds)
 	}
 }
+
+func (r *RateLimitServiceReconciler) getLimitPerTenantFromConfigMap(client k8sclient.Client, ctx context.Context) (uint32, error) {
+
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      multitenantLimitConfigMap,
+			Namespace: r.Namespace,
+		},
+	}
+
+	err := client.Get(context.TODO(), types.NamespacedName{Namespace: r.Namespace, Name: multitenantLimitConfigMap}, configMap)
+	if err != nil && !k8sError.IsNotFound(err) {
+		return 0, fmt.Errorf("Error when getting the config map %w", err)
+	} else if k8sError.IsNotFound(err) {
+		configMap.Data = map[string]string{}
+		_, err = controllerutil.CreateOrUpdate(ctx, client, configMap, func() error {
+			configMap.Data[multitenantRateLimit] = fmt.Sprint(r.getLimitPerTenant())
+			return nil
+		})
```

Comment:
Do you need to check if err != nil here?

Reply:
Good spot! thanks Brian. Updated now.